src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs

```diff
@@ -236,6 +239,24 @@ private protected OptionsBase() { }
             [Argument(ArgumentType.AtMostOnce, HelpText = "Sets the random seed for LightGBM to use.")]
             public int? Seed;
 
+            /// <summary>
+            /// Setting this to true should ensure the stable results when using the same data and the same parameters and different num_threads.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether to use deterministic algorithm.")]
+            public bool Deterministic = false;
```

Comment:
>public bool Deterministic = false; [](http://example.com/codeflow?start=12&length=34)

I know this class is exposing fields directly, but I am wondering can the new added stuff be properties?

Reply:
I believe we use reflection to instantiate these (like when you run this from ML.NET command line), and if its expecting fields it wouldn't work. I will double check on that before I merge this in.

If it is doing that, its possible that we could change things to either do both or only do properites and update all the options as well.